python/src/iceberg/expressions/base.py

```diff
@@ -137,6 +75,28 @@ def __invert__(self) -> "BooleanExpression":
         ...
 
 
+class BoundPredicate(ABC):
+    def __init__(self, left, right):
+        """A concrete predicate must have a `left` and `right`"""
+        self._left = left
+        self._right = right
+
+    @abstractmethod
+    def eval(self, struct: StructProtocol) -> bool:
```

Comment:
It may be misleading to have an eval method when there are so many different ways to evaluate a predicate.

Reply:
I got rid of all the eval methods